packages/mgt-chat/src/components/ChatList/ChatList.tsx

```diff
@@ -132,9 +132,22 @@ export const ChatList = ({
     }
   }, [chatListClient]);
 
+  const markAllThreadsAsRead = (chatThreads: GraphChatThread[]) => {
+    const readChatThreads = chatThreads.map(c => c.id).filter(id => id !== undefined) as string[];
```

Comment:
I don't believe we are checking anywhere else that chatThreads does not have undefined IDs - the solution definitely will fail in a bunch of places if there is no ID.

What I would prefer is wherever chatThreads gets populated we filter out anything that doesn't have an ID so it is never in chatThreads to begin with. For compliance with type checking you can assert id is non-null with id!